src/avram/errors.cr

```diff
@@ -59,4 +59,22 @@ module Avram
       super message
     end
   end
+
+  class ConnectionError < AvramError
+    def initialize(connection_details : URI)
+      error = String.build do |message|
+        message << "Failed to connect to databse '#{connection_details.path.try(&.[1..-1])}' with username #{connection_details.user}.\n"
+        message << "You can try these steps to help:"
+        message << '\n'
+        message << '\n'
+        message << "  ▸ Ensure you have the correct settings in config/database.cr\n"
+        message << "  ▸ Does the database exist? If not, be sure to create it.\n"
```

Comment:
Might be worth including the command `lucky db.create`

Reply:
Also I think this can be shortened. `Be sure the database exists (lucky db.create)`